smart-contracts/contracts/UnlockProtocolGovernor.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import '@openzeppelin/contracts-upgradeable/governance/GovernorUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/governance/extensions/GovernorCountingSimpleUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/governance/extensions/GovernorVotesQuorumFractionUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/governance/extensions/GovernorTimelockControlUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol';
+
+contract UnlockProtocolGovernor is Initializable,
+  GovernorUpgradeable,
+  GovernorCountingSimpleUpgradeable,
+  GovernorVotesUpgradeable,
+  GovernorVotesQuorumFractionUpgradeable,
+  GovernorTimelockControlUpgradeable
+  {
+  function initialize(ERC20VotesUpgradeable _token, TimelockControllerUpgradeable _timelock)
```

Comment:
Turns out I used `ERC20VotesCompUpgradeable` instead of `ERC20VotesUpgradeable` in the UDTv2 contract

https://github.com/unlock-protocol/unlock/blob/6062e8653546f35843640cdda2cf8af1d920cc62/smart-contracts/contracts/UnlockDiscountTokenV2.sol#L13 

Both are similar but the `Comp` version reproduces limitations inherited from Compound suite. From [OZ docs](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#ERC20VotesComp) :
> This version exactly matches Compound’s interface, with the drawback of only supporting supply up to (296 - 1)

Maybe we can upgrade UDT to use the OZ native `ERC20Votes` so we remove that limitation?

Reply:
Ha dang. Yes, probably... Do you think it will break things though?

Reply:
No it seems compatible, plus we can just use Comp in the OZ governor contract. Just it will stop working if the supply gets over what Comp supports

Reply:
Hum. I am worried since the UDT supply is inflationary.  What do you think?

Reply:
The best will be to act now and do a small UDT upgrade replacing the Comp extension by the plain OZ version. Better to pay debts earlier...

Reply:
Agreed@

Reply:
the fix in #7434 will require to go through the mainnet upgrade once again.

Reply:
ok #7434 has been merged